src/WebJobs.Extensions.DurableTask/DurableOrchestrationClient.cs

```diff
@@ -165,12 +158,177 @@ private static JToken ParseToJToken(string value)
+                                    break;
+                                case EventType.EventRaised:
+                                    historyItem.Remove("Input");
+                                    break;
+                                case EventType.OrchestratorStarted:
+                                case EventType.OrchestratorCompleted:
+                                    indexList.Add(i);
+                                    break;
+                            }
+
+                            historyItem.Remove("EventId");
+                            historyItem.Remove("IsPlayed");
+                        }
+                    }
+
+                    var counter = 0;
+                    indexList.Sort();
+                    foreach (var indexValue in indexList)
+                    {
+                        historyArray.RemoveAt(indexValue - counter);
+                        counter++;
+                    }
+                }
+            }
+
+            return new DurableOrchestrationStatus
+            {
+                Name = orchestrationState.Name,
+                InstanceId = orchestrationState.OrchestrationInstance.InstanceId,
+                CreatedTime = orchestrationState.CreatedTime,
+                LastUpdatedTime = orchestrationState.LastUpdatedTime,
+                RuntimeStatus = (OrchestrationRuntimeStatus)orchestrationState.OrchestrationStatus,
+                Input = ParseToJToken(orchestrationState.Input),
+                Output = ParseToJToken(orchestrationState.Output),
+                History = historyArray,
+            };
+        }
+
+        private static void TrackNameAndScheduledTime(JObject historyItem, EventType eventType, int index, Dictionary<string, EventIndexDateMapping> eventMapper)
+        {
+            eventMapper.Add($"{eventType}_{historyItem["EventId"]}", new EventIndexDateMapping { Index = index, Name = (string)historyItem["Name"], Date = (DateTime)historyItem["Timestamp"] });
+        }
+
+        private static void AddScheduledEventDataAndAggregate(ref Dictionary<string, EventIndexDateMapping> eventMapper, string prefix, JToken historyItem, List<int> indexList)
+        {
+            if (!eventMapper.ContainsKey($"{prefix}_{historyItem["TaskScheduledId"]}"))
+            {
+                return;
+            }
+
+            var taskScheduledData = eventMapper[$"{prefix}_{historyItem["TaskScheduledId"]}"];
+            historyItem["ScheduledTime"] = taskScheduledData?.Date;
+            historyItem["FunctionName"] = taskScheduledData?.Name;
+            if (taskScheduledData != null)
+            {
+                indexList.Add(taskScheduledData.Index);
+            }
+        }
+
+        private static void ConvertOutputToJToken(JObject jsonObject, bool showHistoryOutput)
+        {
+            if (!showHistoryOutput)
+            {
+                return;
+            }
+
+            if (jsonObject == null)
```

Comment:
I don't think this null-check is necessary. The code would have failed much earlier than this if `jsonObject` was null.

Reply:
Great optimization! Thank you!